src/pivot-table/hooks/use-column-width.ts

```diff
@@ -87,50 +104,98 @@ export default function useColumnWidth(
-        width = Math.max(
+        const pseudoDimWidth = Math.max(
           ...qMeasureInfo.map(({ qFallbackTitle, columnWidth }) => {
             const fitToContentWidth = measureTextForContent(qFallbackTitle);
             return getColumnWidth(columnWidth, fitToContentWidth);
           }),
         );
+        columnMetadata = {
+          colWidth: pseudoDimWidth,
+          measureTextForHeader: 0,
+          estimateWidthForContent: 0,
+        };
       } else {
         const { qFallbackTitle, qApprMaxGlyphCount, columnWidth } = qDimensionInfo;
-        const fitToContentWidth = Math.max(
-          measureTextForHeader(qFallbackTitle),
-          estimateWidthForRowContent(qApprMaxGlyphCount) + getCollapseExpandIconSize(index),
-        );
 
-        width = getColumnWidth(columnWidth, fitToContentWidth);
+        const measureTextForHeaderResult = measureTextForHeader(qFallbackTitle);
+        const estimateWidthForContentResult =
+          estimateWidthForContent(qApprMaxGlyphCount) + getCollapseExpandIconSize(index);
+        const fitToContentWidth = Math.max(measureTextForHeaderResult, estimateWidthForContentResult);
+
+        columnMetadata = {
+          colWidth: getColumnWidth(columnWidth, fitToContentWidth),
+          measureTextForHeader: measureTextForHeader(qFallbackTitle),
+          estimateWidthForContent: 0,
+        };
       }
 
-      sumOfWidths += width;
-      return width;
+      sumOfWidths += columnMetadata.colWidth;
+      return columnMetadata;
     });
 
     const leftGridMaxWidth = rect.width * LEFT_GRID_MAX_WIDTH_RATIO;
-    if (sumOfWidths < leftGridMaxWidth) return widths;
+    if (sumOfWidths < leftGridMaxWidth) return allColumnsMetadata;
 
     const multiplier = leftGridMaxWidth / sumOfWidths;
-    return widths.map((w) => w * multiplier);
+    return allColumnsMetadata.map((col) => ({ ...col, colWidth: col.colWidth * multiplier }));
   }, [
     visibleLeftDimensionInfo,
     measureTextForHeader,
-    estimateWidthForRowContent,
+    estimateWidthForContent,
     getCollapseExpandIconSize,
     rect.width,
     qMeasureInfo,
     measureTextForContent,
   ]);
 
-  const getLeftGridColumnWidth = useCallback((index: number) => leftGridColumnWidths[index], [leftGridColumnWidths]);
+  const getLeftGridColumnWidth = useCallback(
+    (index: number) => leftGridColumnWidths[index].colWidth,
+    [leftGridColumnWidths],
+  );
+
+  const getLeftGridColumnWidthMetadata = useCallback<GetLeftColumnWidthMetadata>(
+    (idx, isLocked) => {
+      const metaData = leftGridColumnWidths[idx];
+      let shouldShowMenuIcon = true;
+      let shouldShowLockIcon = true;
+
+      // margin left + icon size
+      const lockIconWidth = isLocked ? 8 + 12 : 0;
+      // size + grid gap
+      const menuIconWidth = 24 + 4;
+      // 8px padding left, 4px padding right
+      let finalSize = lockIconWidth + 8 + metaData.measureTextForHeader + 4 + menuIconWidth - LEEWAY_WIDTH;
```

Comment:
I have my 👀's on the LEEWAY_WIDTH, depending on which pr goes in first, i'll take care of it

Reply:
It's gone and replaced by `TOTAL_CELL_WIDTH`

Reply:
For this, we need to get your PR in first!
since you are using `useMeasureText` from shared repo
and those changes will get into main by your PR

Reply:
It's in!

Reply:
I was thinking about this one: https://github.com/qlik-oss/sn-pivot-table/pull/388